### DISCUSSION TITLE
ReCreation of Kubernetes Clusters

### DISCUSSION THREAD

**Discussion by domgreen** [fluxcd/flux2#257] [Q&A] [answered]
I regularly destroy and recreate my clusters via Terraform is there a pattern or best practice to automate the resuming of reconciliation after the recreation of a cluster?

At the moment I am using bootstrap and then re-adding the source... is this the general pattern?

**Comment by stefanprodan** [comment, marked as answer]
> At the moment I am using bootstrap and then re-adding the source... is this the general pattern?

Yes, to make this easier we could provide a terraform module that downloads the gotk binary and runs `gotk bootstrap` on the target cluster. In the TF module we could expose the bootstrap params such as components, version, git repo, etc.

**Reply by domgreen** [reply]
Actually at the moment just bootstrap doesnt work ... I need to apply the yaml from `fleet-infra/blah/gitops-system` manually.

**Reply by stefanprodan** [reply]
Can you paste the bootstrap logs here please, it should apply those files on its own.

**Reply by domgreen** [reply]
Any extra flags I should add? 

Will do this tomorrow and paste back as I'm out of the office for the rest of the day 👍

**Reply by stefanprodan** [reply]
I've found the issue, fixed in: https://github.com/fluxcd/toolkit/pull/259
